tests/puppeteer/pages/BO/BO_login.js

```diff
@@ -0,0 +1,31 @@
+const Page = require('../page');
+
+module.exports = class BO_login extends Page {
+
+  constructor() {
+    super();
+
+    this.BO_login_pageTitle = 'Dashboard • PrestaShop';
+
+    this.BO_login_email_input = "#email";
+    this.BO_login_password_input = "#passwd";
+    this.BO_login_submitLogin_button = "#submit_login";
+    this.BO_login_stayLoggedIn_checkbox = "#stay_logged_in";
+
+    this.BO_login_forgotPassword_link = "#a.show-forgot-password";
+    this.BO_login_emailForgot_input = "#email_forgot";
+    this.BO_login_sendResetLink_button = "#button[name=submitLogin].btn-default";
+  }
+
+  /*
+  Methods
+   */
+
+  async login(email, passwd) {
+    await global.page.type(this.BO_login_email_input, email);
```

Comment:
Can we avoid `global` variables and use Dependency Injection instead ?

Global variables can lead to crazy buggy situations when one part of the code modifies it and the other one depends on it _and you have no clue who did this_ 😛 and also are a bit of code smell as it promotes lazy variables sharing.

Dependency Injection will protect you from that 😉

Reply:
Resolved, thanks !